crates/miden-testing/src/kernel_tests/tx/test_output_note.rs

```diff
@@ -0,0 +1,470 @@
+            # => [num_assets_0, note_idx]
+
+            # assert the number of assets
+            push.{assets_number_0}
+            assert_eq.err="number of assets in the note having only asset_0 is incorrect"
+            # => [note_idx]
+
+            # get the assets info once more to get the cached data and assert that this data didn't
+            # change
+            dup exec.output_note::get_assets_info
+            push.{COMPUTED_ASSETS_COMMITMENT_0}
+            assert_eqw.err="assets commitment of the note having only asset_0 is incorrect"
+            push.{assets_number_0}
+            assert_eq.err="number of assets in the note having only asset_0 is incorrect"
+            # => [note_idx]
+
+            # add asset_1 to the note
+            push.{asset_1}
+            call.::miden::contracts::wallets::basic::move_asset_to_note
+            dropw
+            # => [note_idx]
+
+            # get the assets hash and assets number of the note having asset_0 and asset_1
+            dup exec.output_note::get_assets_info
+            # => [ASSETS_COMMITMENT_1, num_assets_1, note_idx]
+
+            # assert the correctness of the assets hash
+            push.{COMPUTED_ASSETS_COMMITMENT_1}
+            assert_eqw.err="assets commitment of the note having asset_0 and asset_1 is incorrect"
+            # => [num_assets_1, note_idx]
+
+            # assert the number of assets
+            push.{assets_number_1}
+            assert_eq.err="number of assets in the note having asset_0 and asset_1 is incorrect"
+            # => [note_idx]
+
+            # truncate the stack
+            exec.sys::truncate_stack
+        end
+        "#,
+        // output note
+        output_note = create_output_note_with_asset(
+            &output_note_1.recipient().digest(),
+            output_note_1.metadata().execution_hint(),
+            NoteType::Public,
+            output_note_1.metadata().tag(),
+            &fungible_asset_0.into()
+        ),
+        // first data request
+        COMPUTED_ASSETS_COMMITMENT_0 =
+            word_to_masm_push_string(&output_note_0.assets().commitment()),
+        assets_number_0 = output_note_0.assets().num_assets(),
+        // second data request
+        asset_1 = word_to_masm_push_string(&fungible_asset_1.into()),
+        COMPUTED_ASSETS_COMMITMENT_1 =
+            word_to_masm_push_string(&output_note_1.assets().commitment()),
+        assets_number_1 = output_note_1.assets().num_assets(),
+    );
+
+    let tx_script =
+        TransactionScript::compile(tx_script_src, TransactionKernel::testing_assembler())?;
+
+    let tx_context = mock_chain
+        .build_tx_context(account.id(), &[], &[])?
+        .extend_expected_output_notes(vec![OutputNote::Full(output_note_1)])
+        .tx_script(tx_script)
+        .build()?;
+
+    tx_context.execute()?;
+
+    Ok(())
+}
+
+#[test]
+fn test_get_recipient_and_metadata() -> anyhow::Result<()> {
+    let mut builder = MockChain::builder();
```

Comment:
Same question here as for input notes. Do we get additional coverage by testing with two notes rather than one?

Reply:
You're right, there is no benefit of checking several notes compared to just one. I'll rework both test.